### DISCUSSION TITLE
Tests: PulseTemplateStub and DummyPulseTemplate duplicates?

### DISCUSSION THREAD

**Discussion by lumip** [qutech/qupulse#729] [General]

**Comment by shumpohl** [comment]
In a sense yes. But the stubs throw an exception if their properties are accessed. At least, this was the idea as far as I remember.

**Comment by lumip** [comment]
Is that required?

**Comment by shumpohl** [comment]
The properties should not be used during the tests.

The purpose of the PulseTemplateStub is to test the code of the PulseTemplate class itself. The purpose of the DummyPulseTemplate is to provide a dummy for other tests.
